src/btree/bt_walk.c

```diff
@@ -387,18 +387,13 @@ __tree_walk_internal(WT_SESSION_IMPL *session, WT_REF **refp, uint64_t *walkcntp
              * not expected, our parent WT_REF should not have split.
              */
             ret = __wt_page_swap(session, couple, ref, WT_READ_NOTFOUND_OK | flags);
+            WT_ERR_NOTFOUND_KEEP(ret);
             if (ret == 0) {
                 /* Success, "couple" released. */
                 couple = NULL;
                 *refp = ref;
                 goto done;
             }
-
-            /* An expected error, so "couple" is unchanged. */
-            if (ret == WT_NOTFOUND)
-                continue;
```

Comment:
This seems surprising. Oh, I see it gets to the end of the control loop which is the same as the continue used to be. The code structure worries me a little here now - it seems prone to breaking if someone adds code after the `if` statement. Perhaps switching to be `if (ret == WT_NOTFOUND) continue;` would make that less likely.

Reply:
As in, you think we should leave in the original lines? Probably with a comment. I wonder if Coverity would then flag it.

Reply:
We could also put a comment there without the original lines.  Something like: `ret must be WT_NOTFOUND, an expected error, continue with "couple" unchanged.` It would serve the same goal of informing people in the future what's going on there.